### DISCUSSION TITLE
Disable automated failover

### DISCUSSION THREAD

**Discussion by francoispqt** [cloudnative-pg/cloudnative-pg#466] [Ideas]
Currently there is no way to disable automated failover. I think this is something we should be able to do, either through a primary failure time settings (primary needs to be failing since more than x seconds before triggering automated failover) or just a disable failover boolean.

**Comment by gbartolini** [comment]
First, you have no idea how happy I am to receive this request - which I have transformed in a discussion.

The reason is that it means that our operator is getting where we had initially thought of, and we were hoping to get to this point at some stage.

We obviously prioritized the automated failover process in the first place, rather than data loss (so RTO over RPO). The reason is that, through syncrep, you can mitigate the risk of data loss. But there are obviously cases where the administrator can request to wait to failover and prefers to wait for the primary to come back up.

We can definitely add a timeout/delay to control this (at the moment this timeout/delay is 0). If you read the [automated failover](https://cloudnative-pg.io/documentation/1.16/failover/) section, you see that for now we have only implemented the switchoverDelay parameter.

We could think of a similar parameter that prolongs the failover detection range from the current 0 value to a specified delay time.

Does this resonate with you?

**Reply by francoispqt** [reply]
It definitely makes sense. I completely understand you prioritized on the automated failover process.

Yes the timeout/delay settings to trigger a failover is what we are looking for as opposed to the switchoverDelay which is a graceful timeout given to the primary node to terminate once the switchover process has already triggered. I'd be happy to work on it.

**Reply by gbartolini** [reply]
This is great. Please feel free to propose a PR. Thanks!

**Reply by gbartolini** [reply]
Just for the record - this is the proposed PR: https://github.com/cloudnative-pg/cloudnative-pg/pull/508
